src/movies.js

```diff
@@ -1,30 +1,186 @@
+    return result.director === 'Steven Spielberg' && result.genre.includes('Drama') === true;
+  })
+  return totStevenMovies.length;
+}
+howManyMovies(movies);
 
-// Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+// Iteration 3: All scores average - Get the average of all scores with 2 Decimals
+function scoresAverage(moviesArr) {
+  if (moviesArr == false) {
+    return 0;
+  }
+  // [ 1  , 2,  3]
+  const totNumDec = moviesArr.reduce((result, valPos) =>{
+    if (valPos.score) {
+      return result + valPos.score;
+    }else {
+      return result;
+    }
+  }, 0);
+  return Math.round((totNumDec / moviesArr.length) * 100) / 100;
+}
+scoresAverage(movies);
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(moviesArr) {
+  if (moviesArr == false) {
+    return 0;
+  }
+  let count = 0;
+  const totNumDec = moviesArr.reduce((result, valPos) =>{
+    // console.log(result);
+    if (valPos.genre.includes('Drama') === true) {
+      count++;
+      return result + valPos.score;
+    }else {
+      return result;
+    }
+  }, 0);
+  if (count == 0) {
+    return 0;
+  }else {
+    return Math.round((totNumDec / count) * 100) / 100;
+  }
+}
+dramaMoviesScore(movies);
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(moviesArr) {
+  if (!moviesArr.length) {
+    return 0;
+  }
+  const segArr = moviesArr.sort((a, b) => {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    }
+    return a.year - b.year;
+  });
+  return segArr;
+}
+orderByYear(movies);
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(moviesArr) {
+  const arrOrder = moviesArr
+  .map((elemento) => {
+    return elemento.title;
+  })
+  .sort((a, b) => {
+    return a.localeCompare(b);
+  });
+  const arrFinal = arrOrder.slice(0, 20);
+  return arrFinal;
+}
+orderAlphabetically(movies);
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes() {}
+function turnHoursToMinutes(moviesArr) {
+  let testArr = [];
+  const arrStrgToArr = moviesArr.map( element => {
+  let arrStr = element.duration.split(' ');
+  if (arrStr.length == 2) {
+    return parseFloat(arrStr[0], 10) * 60 + parseFloat(arrStr[1], 10);
+  }else if (arrStr[0].indexOf('min') == -1) {
+    return parseFloat(arrStr[0], 10) * 60 ;
+  }else {
+    return parseFloat(arrStr[0], 10);
+  }
+  });
+  for (let i = 0; i < moviesArr.length; i++) {
+    testArr.push(moviesArr);
+    testArr[i].duration = arrStrgToArr[i];
+  }
+  return testArr;
+}
+turnHoursToMinutes(movies);
 
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg() {}
-
+function bestYearAvg(moviesArr) {
```

Comment:
Esta es una opción: ```suggestion
function bestYearAvg(lotsOfMovies) {
    if (!lotsOfMovies.length) return null;
  
    let masterObject = {};
  
    lotsOfMovies.forEach(eachMovie => {
      if (!masterObject[eachMovie.year]) {
        masterObject[eachMovie.year] = [eachMovie];
        //console.log("------>",masterObject[eachMovie.year])
      } else {
        masterObject[eachMovie.year].push(eachMovie);

      }
    });
  console.log("this is master",masterObject)
    let highest = 0;
    let theActualYear;
    for (let theYear in masterObject) {
      if (scoresAverage(masterObject[theYear]) > highest) {
        highest = scoresAverage(masterObject[theYear]);
        theActualYear = theYear;
      }
    }
    return `The best year was ${theActualYear} with an average rate of ${highest}`;
  }
 bestYearAvg(movies) 
```suggestion
function bestYearAvg(moviesArr) {
```

Reply:
Lo que esta sucediendo es que en un objeto itera las peliculas y guarda en el cada años y a ese año le agrega todas las peliculas que le corresponden mas adelante reutilizamos tu función de promedio para ir evaluando cual es mayor y para eso hacemos uso de un contador en 0 mientras va iterando iguala al que es más alto